test/src/util/pelz_json_parser_test_suite.c

```diff
@@ -0,0 +1,440 @@
+/*
+ * pelz_json_parser_suite.c
+ */
+
+#include "pelz_json_parser_test_suite.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <cjson/cJSON.h>
+
+#include <CharBuf.h>
+#include <pelz_log.h>
+
+// Adds all key table tests to main test runner.
+int pelz_json_parser_suite_add_tests(CU_pSuite suite)
+{
+  if (NULL == CU_add_test(suite, "Test JSON Encryption Request Parser", test_encrypt_parser))
+  {
+    return (1);
+  }
+  if (NULL == CU_add_test(suite, "Test JSON Decryption Request Parser", test_decrypt_parser))
+  {
+    return (1);
+  }
+  if (NULL == CU_add_test(suite, "Test Decoding of JSON formatted Request", test_request_decoder))
+  {
+    return (1);
+  }
+  if (NULL == CU_add_test(suite, "Test Encoding of JSON formatted Response Message", test_message_encoder))
+  {
+    return (1);
+  }
+  if (NULL == CU_add_test(suite, "Test Encoding of JSON formatted Error Message", test_error_message_encoder))
+  {
+    return (1);
+  }
+  return (0);
+}
+
+void test_encrypt_parser(void)
+{
+  cJSON *json;
+  CharBuf key_id;
+  CharBuf data;
+
+  //Valid Test Values
+  char *json_key_id = "file:/test/key1.txt";
+  int json_key_id_len = 19;
+  char *enc_data = "QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVoxMjM0NTY=\n";
+  int enc_data_len = 45;
+  char *dec_data = "SwqqSZbNtN2SOfKGtE2jfklrcARSCZE9Tdl93pggkIsRkY3MrjevmQ==\n";
+  int dec_data_len = 57;
+
+  //Building of a standard valid JSON request
+  json = cJSON_CreateObject();
+  cJSON_AddItemToObject(json, "request_type", cJSON_CreateNumber(1));
+  cJSON_AddItemToObject(json, "key_id", cJSON_CreateString(json_key_id));
+  cJSON_AddItemToObject(json, "key_id_len", cJSON_CreateNumber(json_key_id_len));
+  cJSON_AddItemToObject(json, "enc_data", cJSON_CreateString(enc_data));
+  cJSON_AddItemToObject(json, "enc_data_len", cJSON_CreateNumber(enc_data_len));
+  cJSON_AddItemToObject(json, "dec_data", cJSON_CreateString(dec_data));
+  cJSON_AddItemToObject(json, "dec_data_len", cJSON_CreateNumber(dec_data_len));
+
+  //Test standard valid JSON request
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 0);
```

Comment:
Should we also check that the correct values got put into `key_id` and `data` here?